packages/core-forger/src/client.ts

```diff
@@ -145,40 +107,43 @@ export class Client {
             return this.logger.error("Was unable to find any local hosts.");
         }
 
-        try {
-            await this.__post(`${host}/internal/utils/events`, { event, body });
-        } catch (error) {
-            this.logger.error(`Failed to emit "${event}" to "${host}"`);
-        }
+        await this.post(`${host}/internal/utils/events`, { event, body });
     }
 
     /**
      * Chose a responsive host.
-     * @return {void}
      */
-    public async __chooseHost(wait = 0) {
-        const host = sample(this.hosts);
-
-        try {
-            await this.__get(`${host}/peer/status`);
-
-            this.host = host;
-        } catch (error) {
-            this.logger.debug(`${host} didn't respond to the forger. Trying another host`);
-
-            if (wait > 0) {
-                await delay(wait);
+    public async selectHost(): Promise<void> {
+        let queriedHosts = 0;
+        for (const host of this.hosts) {
+            try {
+                await this.get(`${host}/peer/status`);
+                this.host = host;
+            } catch (error) {
+                if (queriedHosts === this.hosts.length - 1) {
+                    throw new HostNoResponseError(host);
+                } else {
+                    this.logger.warn(`Failed to get response from ${host}. Trying another host.`);
+                }
+            } finally {
+                queriedHosts++;
             }
-
-            await this.__chooseHost(wait);
         }
     }
 
-    public async __get(url, timeout: number = 2000) {
-        return httpie.get(url, { headers: this.headers, timeout });
+    private async get(url, timeout: number = 2000) {
```

Comment:
Should add a struct/interface for the expected return value, maybe add a `HttpieResponse` to `@arkecosystem/core-utils`.